[Bug]: Null Pointer Exception when Pushing to Remote Respository

### Apache Hop version?

2.7

### Java version?

Oracle JDK 11.0.19

### Operating system

Windows

### What happened?

Within hop-gui, after committing code to a local repository, the user pushes to a GitLab remote server using the Push icon within the File Explorer and the following exception appears:

```
HopException: 
There was an error doing a git push
git@<SERVERNAME>:<STUB>/<GIT REPO NAME>.git: remote hung up unexpectedly

Root cause: NullPointerException: 

When clicking Details for the complete trace, the following appears:

org.apache.hop.core.exception.HopException: 
There was an error doing a git push
git@<SERVERNAME>:<STUB>/<GIT REPO NAME>.git: remote hung up unexpectedly

	at org.apache.hop.git.model.UIGit.push(UIGit.java:582)
	at org.apache.hop.git.model.UIGit.push(UIGit.java:528)
	at org.apache.hop.git.GitGuiPlugin.gitPush(GitGuiPlugin.java:227)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.hop.ui.core.gui.BaseGuiWidgets.lambda$getListener$1(BaseGuiWidgets.java:216)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4256)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4054)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3642)
	at org.apache.hop.ui.hopgui.HopGui.open(HopGui.java:473)
	at org.apache.hop.ui.hopgui.HopGui.main(HopGui.java:351)
Caused by: org.eclipse.jgit.api.errors.TransportException: git@<SERVERNAME>:<STUB>/<GIT REPO NAME>.git: remote hung up unexpectedly
	at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:168)
	at org.apache.hop.git.model.UIGit.push(UIGit.java:570)
	... 14 more
Caused by: org.eclipse.jgit.errors.TransportException: git@<SERVERNAME>:<STUB>/<GIT REPO NAME>.git: remote hung up unexpectedly
	at org.eclipse.jgit.transport.TransportGitSsh$SshPushConnection.<init>(TransportGitSsh.java:380)
	at org.eclipse.jgit.transport.TransportGitSsh.openPush(TransportGitSsh.java:157)
	at org.eclipse.jgit.transport.PushProcess.execute(PushProcess.java:140)
	at org.eclipse.jgit.transport.Transport.push(Transport.java:1540)
	at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:158)
	... 15 more
Caused by: java.lang.NullPointerException
	at org.eclipse.jgit.transport.SshTransport.getSession(SshTransport.java:107)
	at org.eclipse.jgit.transport.TransportGitSsh$SshPushConnection.<init>(TransportGitSsh.java:356)
	... 19 more

```
 

### Issue Priority

Priority: 3

### Issue Component

Component: Hop Gui

--- a/plugins/misc/git/pom.xml
+++ b/plugins/misc/git/pom.xml
@@ -29,7 +29,7 @@
     <name>Hop Plugins Miscellaneous Git</name>
 
     <properties>
-        <dependency.jgit.revision>6.10.0.202406032230-r</dependency.jgit.revision>
+        <dependency.jgit.revision>7.3.0.202506031305-r</dependency.jgit.revision>
     </properties>
 
     <dependencyManagement>
@@ -53,6 +53,11 @@
         <dependency>
             <groupId>org.eclipse.jgit</groupId>
             <artifactId>org.eclipse.jgit.http.apache</artifactId>
+            <version>${dependency.jgit.revision}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.jgit</groupId>
+            <artifactId>org.eclipse.jgit.ssh.apache</artifactId>
             <version>${dependency.jgit.revision}</version>
         </dependency>
         <dependency>

--- a/plugins/misc/git/src/main/java/org/apache/hop/git/model/UIGit.java
+++ b/plugins/misc/git/src/main/java/org/apache/hop/git/model/UIGit.java
@@ -33,6 +33,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
 import org.apache.hop.core.Const;
 import org.apache.hop.core.exception.HopException;
 import org.apache.hop.core.exception.HopFileException;
@@ -92,9 +93,12 @@ import org.eclipse.jgit.transport.PushResult;
 import org.eclipse.jgit.transport.RefSpec;
 import org.eclipse.jgit.transport.RemoteConfig;
 import org.eclipse.jgit.transport.RemoteRefUpdate;
+import org.eclipse.jgit.transport.SshSessionFactory;
 import org.eclipse.jgit.transport.URIish;
 import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
 import org.eclipse.jgit.transport.http.apache.HttpClientConnectionFactory;
+import org.eclipse.jgit.transport.sshd.SshdSessionFactory;
+import org.eclipse.jgit.transport.sshd.SshdSessionFactoryBuilder;
 import org.eclipse.jgit.treewalk.AbstractTreeIterator;
 import org.eclipse.jgit.treewalk.CanonicalTreeParser;
 import org.eclipse.jgit.treewalk.EmptyTreeIterator;
@@ -592,11 +596,26 @@ public class UIGit extends VCS {
     try {
       name = name == null ? null : getExpandedName(name, type);
 
-      PushCommand cmd = git.push();
-      cmd.setCredentialsProvider(credentialsProvider);
+      PushCommand cmd;
+
+      String url = git.getRepository().getConfig().getString("remote", "origin", "url");
+      if (!StringUtils.isEmpty(url) && (url.startsWith("https://") || url.startsWith("http://"))) {
+        cmd = git.push();
+        cmd.setCredentialsProvider(credentialsProvider);
+      } else {
+        SshdSessionFactory customFactory =
+            new SshdSessionFactoryBuilder()
+                .setHomeDirectory(new File(System.getProperty("user.home")))
+                .setSshDirectory(new File(System.getProperty("user.home"), ".ssh"))
+                .build(null);
+        SshSessionFactory.setInstance(customFactory);
+        cmd = git.push();
+      }
+
       if (name != null) {
         cmd.setRefSpecs(new RefSpec(name));
       }
+
       Iterable<PushResult> resultIterable = cmd.call();
       processPushResult(resultIterable);
       return true;

